### PR TITLE
Drop support for Python 3.9, NumPy 1.22 and 1.23, SciPy 1.7, 1.8 and 1.9

### PR DESCRIPTION
**Pull request by asi1024**
Close https://github.com/cupy/cupy/issues/8484.

### CODE REVIEW

**Comment by kmaehashi** [comment]
For RHEL (CentOS) 7 you need these lines to build Python 3.10:
https://github.com/cupy/cupy-release-tools/blob/805ff50d47a33ddae9e73754e06d4865890bdac3/builder/setup_python.sh#L7-L12

File: .pfnci/generate.py [hunk 1]
```diff
@@ -130,6 +130,14 @@ def generate_dockerfile(self) -> str:
                     '',
                 ]
 
+                if matrix.python == "3.10":
+                    # CentOS 7: Use OpenSSL 1.1 for Python 3.10+.
+                    lines += [
+                        'ENV CFLAGS "-I/usr/include/openssl11"',
+                        'ENV CPPFLAGS "-I/usr/include/openssl11"',
+                        'ENV LDFLAGS "-L/usr/lib64/openssl11"',
+                    ]
```
Comment (asi1024) [inline .pfnci/generate.py:139]: https://ci.preferred.jp/cupy.linux.cuda112/167673/

**Review by kmaehashi — APPROVED** [review]

**Pull request merged**